ubi_manifest/worker/tasks/content_audit.py

```diff
@@ -0,0 +1,146 @@
+import logging
+import os
+
+from pubtools.pulplib import Criteria
+
+from ubi_manifest.worker.tasks.celery import app
+from ubi_manifest.worker.tasks.depsolve import filter_whitelist, get_content_config
+from ubi_manifest.worker.tasks.depsolver.models import UbiUnit
+from ubi_manifest.worker.tasks.depsolver.pulp_queries import (
+    search_modulemd_defaults,
+    search_modulemds,
+    search_rpms,
+)
+from ubi_manifest.worker.tasks.depsolver.ubi_config import UbiConfigLoader
+from ubi_manifest.worker.tasks.depsolver.utils import (
+    create_or_criteria,
+    get_criteria_for_modules,
+    is_blacklisted,
+    keep_n_latest_modulemd_defaults,
+    keep_n_latest_modules,
+    keep_n_latest_rpms,
+    make_pulp_client,
+    parse_blacklist_config,
+)
+
+_LOG = logging.getLogger(__name__)
+MAX_WORKERS = int(os.getenv("UBI_MANIFEST_CONTENT_AUDIT_WORKERS", "8"))
+
+
+@app.task  # type: ignore [misc]  # ignore untyped decorator
+def content_audit_task() -> None:
+    """
+    This task checks that all available content is up-to-date, that whitelisted
+    content is present, and that blacklisted content is absent.
+    """
+
+    config_loaders = [UbiConfigLoader(url) for url in app.conf.content_config.values()]
+
+    with make_pulp_client(app.conf) as client:
+        for out_repo in client.search_repository(
+            Criteria.with_field("ubi_population", True)
+        ):
+            # get all relevant units currently on output repo
+            out_rpms = search_rpms([Criteria.true()], [out_repo]).result()
+            out_mds = search_modulemds([Criteria.true()], [out_repo]).result()
+            out_mdds = search_modulemd_defaults([Criteria.true()], [out_repo]).result()
+
+            seen_units = set()
+            output_whitelist = set()
+
+            for in_repo in client.search_repository(
```

Comment:
Sorry it this wasn't clear previously, but you need to get the latest version from all `population_sources`.
I believe this should even make thing easier.

So in the code below you likely need to do:
```
                in_rpms_ft = search_rpms(
                    create_or_criteria(["name"], [(rpm.name,) for rpm in out_rpms]),
                    ----> in_repos, <-----
                )
```